### PR TITLE
enh(core): gorgone add a source install script

### PR DESCRIPTION
**Pull request by sc979**
Add an installation from sources script

### CODE REVIEW

File: sourceInstall/functions [hunk 1]
```diff
@@ -0,0 +1,1004 @@
...
+			    [ -d "$res" ] && break
+			    yes_no_default "Do you want to create this directory ? [$res]"
+			fi
+			if [ $? -eq 0 ] || [ $CREATION_ALLOWED -eq 1 ] ; then
+			    mkdir -p $res
+			    log "INFO" "Creating $res"
+			elif [ $? -ne 0 ] ; then
+			    echo_passed "Could not create directory." "$critical"
+			    #continue
+            fi
```
Comment (sc979) [inline sourceInstall/functions]: wrong indentation

File: sourceInstall/vars [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+# sources and log paths
+LOG_DIR="$BASE_DIR/log"
+LOG_FILE="$LOG_DIR/gorgone_install.log"
+
+# deployment paths
+DEFAULT_GORGONE_LOG="/var/log/centreon-gorgone"
+DEFAULT_GORGONE_VARLIB="/var/lib/centreon-gorgone"
+DEFAULT_GORGONE_ETC="/etc/centreon-gorgone"
+DEFAULT_GORGONE_CONF_FILE="/config.yaml"
+DEFAULT_GORGONE_BINDIR="/usr/local/bin/"
```
Comment (cgagnaire) [inline sourceInstall/vars]: To match systemd config file:
```suggestion
DEFAULT_GORGONE_BINDIR="/usr/bin/"
```

File: install.sh [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+echo -e "\tInstalling Gorgone daemon"
+echo "$line"
+
+## Copy the files in destination folders and modify rights
+copy_and_modify_rights "$BASE_DIR/config/systemd" "gorgoned-service" "$SYSTEM_D" "gorgoned.service" "664"
+copy_and_modify_rights "$BASE_DIR/config/systemd" "gorgoned-sysconfig" "$SYSCONFIG" "gorgoned" "664"
+copy_and_modify_rights "$BASE_DIR/config/logrotate" "gorgoned" "$LOGROTATE_D" "gorgoned" "775"
+copy_and_modify_rights "$BASE_DIR/packaging" "config.yaml" "$GORGONE_ETC" "config.yaml" "755"
+copy_and_modify_rights "$BASE_DIR" "gorgoned" "$GORGONE_BINDIR" "gorgoned" "755"
+copy_and_modify_rights "$BASE_DIR/contrib" "gorgone_config_init.pl" "/usr/local/bin" "gorgone_config_init.pl" "775"
```
Comment (cgagnaire) [inline install.sh]: Use bin dir defined by user?
```suggestion
copy_and_modify_rights "$BASE_DIR/contrib" "gorgone_config_init.pl" "$GORGONE_BINDIR" "gorgone_config_init.pl" "775"
```

File: install.sh [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+echo -e "$line"
+
+## check filesystem space
+check_disk_space
+
+## define destination folders
+allow_creation_of_missing_folders
+locate_gorgone_logdir
+locate_gorgone_varlib
+locate_gorgone_etcdir
```
Comment (cgagnaire) [inline install.sh]: Those directories need Gorgone's user and group rights.

Comment (sc979) [reply install.sh]: I'll find a way to do so

File: install.sh [hunk 5]
```diff
@@ -0,0 +1,274 @@
...
+
+## create gorgone user
+check_gorgone_group
+check_gorgone_user
+
+echo -e "\n$line"
+echo -e "\tAdding Gorgone user to the mandatory folders"
+echo -e "$line"
+
+change_rights "$GORGONE_USER" "$GORGONE_GROUP" "775" "$LOG_DIR"
```
Comment (cgagnaire) [inline install.sh]:
```suggestion
change_rights "$GORGONE_USER" "$GORGONE_GROUP" "775" "$GORGONE_LOG"
```

**Pull request merged**